addons/account_edi/views/account_move_views.xml

```diff
@@ -96,6 +97,12 @@
                         invisible="edi_error_count == 0 or edi_blocking_level != 'error'">
                         <div class="o_row">
                             <field name="edi_error_message" />
+                            <button name="force_cancel_edi"
+                                string="Force EDI Cancellation"
+                                type="object"
+                                groups="account.group_account_invoice"
+                                confirm="Are you sure you want to force cancel the EDI document?"
+                                invisible="not (edi_show_force_cancel and edi_show_abandon_cancel_button)" />
```

Comment:
```suggestion
                                invisible="not edi_show_force_cancel_button"/>
```
I would make the visibility depend on just one field, and make that field's compute logic handle everything. Easier to customize when inheriting.